### PR TITLE
85 search for friends modal frontend

### PR DESCRIPTION
**Pull request by dorianjagusch**
The frontend has the search friends modal. 
This means we can create friendships from the frontend now!

## Changes
- Added Search friends on the frontend
- Modal and Form creation
- minor refactorings on some frontend helper functions
- Added some TODOs to keep track of things to change in the near future

## Caveats
- While the name specific search is ready on the frontend, the backend needs to handle search by username and return the relationship with the user making the request #201 
- Once the image PR #200 is through, have frontend fetch the pictures of search results

### CODE REVIEW

File: frontend/static/js/components/dialogs/authenticationModal.js [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 
 export default class AuthenticationModal extends ADialog {
 	constructor(parentCallback) {
 		super(new loginForm(), new AuthenticationService());
-		this.notify = this.notify.bind(this);
 		this.getFormData = this.getFormData.bind(this);
 		this.authenticateUser = this.authenticateUser.bind(this);
-		this.getFormData = this.getFormData.bind(this);
-		this.onDataReceived = parentCallback;
+		this.onDataReceived = parentCallback;8;
```
Comment (merituulie) [inline frontend/static/js/components/dialogs/authenticationModal.js:10]: whats with the 8:?

File: frontend/static/js/components/dialogs/tournamentModals/selectPlayersModal.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		const tournamentData = {
+			players,
+			tournamentId: this.tournamentId,
+			tournamentName: this.tournamentName,
+		};
+		return tournamentData;
+	}
+
+	appendEventlistenters() {
+		this.dialog.addEventListener(
```
Comment (merituulie) [inline frontend/static/js/components/dialogs/tournamentModals/selectPlayersModal.js:95]: mmeh, quite a lot of nesting here, but I guess its the javascript way.. might make sense to separate, but low priority

File: frontend/static/js/components/formComponents/selectPlayersForm.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+		const playerList = document.createElement('section');
+		playerList.classList.add('player-list', `player-${this.numberOfPlayers}`);
+		form.appendChild(playerList);
+
+		for (let i = 0; i < this.numberOfPlayers; i++) {
+			if (i == 0) {
+				const playerCard = PlayerCard(
+					localStorage.getItem('username'),
+					// localStorage.getItem('userImage')
```
Comment (merituulie) [inline frontend/static/js/components/formComponents/selectPlayersForm.js:25]: same here, we would actually get the profile picture from backend at this point!

File: frontend/static/js/components/formComponents/summaryForm.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import AForm from "./AForm.js";7
```
Comment (merituulie) [inline frontend/static/js/components/formComponents/summaryForm.js:1]: 7 here? what is this sorcery

**Review by merituulie — DISMISSED** [review]

File: test.html [hunk 5]
```diff
@@ -0,0 +1,121 @@
+<!DOCTYPE html>
```
Comment (merituulie) [inline test.html:1]: who dis

**Comment by dorianjagusch** [comment]
I changed the branch this merges into. Sorry for the oversight. The PR is not as bloated as the tournament modal one :see_no_evil:

**Review by sakarisalmi — APPROVED** [review]
Nice!

**Review by merituulie — APPROVED** [review]

**Pull request merged**